model/task/db.go

```diff
@@ -766,6 +766,32 @@ func schedulableHostTasksQuery() bson.M {
 	return q
 }
 
+// FindNeedsContainerAllocation returns all container tasks that are waiting for
+// a container to be allocated to them sorted by activation time.
+func FindNeedsContainerAllocation() ([]Task, error) {
+	q := bson.M{
+		ActivatedKey:         true,
+		StatusKey:            evergreen.TaskContainerUnallocated,
+		ExecutionPlatformKey: ExecutionPlatformContainer,
+		PriorityKey:          bson.M{"$gt": evergreen.DisabledTaskPriority},
+		"$or": []bson.M{
+			{
+				DependsOnKey: bson.M{"$size": 0},
+			},
+			{
+				// Containers can only be allocated for tasks whose dependencies
+				// are all met. All dependencies are met if they're all finished
+				// running and are still attainable (i.e. the dependency's
+				// required status matched the task's actual ending status).
+				bsonutil.GetDottedKeyName(DependsOnKey, DependencyFinishedKey):     true,
+				bsonutil.GetDottedKeyName(DependsOnKey, DependencyUnattainableKey): bson.M{"$ne": true},
+			},
+			{OverrideDependenciesKey: true},
+		},
+	}
+	return FindAll(db.Query(q).Sort([]string{ActivatedTimeKey}))
```

Comment:
Since this queue is in-memory, how do we ensure that multiple app servers do not pick up the same tasks? Should this be a transaction to ensure that we update the `StatusKey`?

Reply:
Amboy scopes are going to resolve any issues with duplicate pod allocator jobs being created from these tasks.

We don't need to update the status key since this is just the query. The status is only updated from unallocated to allocated by the individual pod allocator jobs. This task queue will just feed into the cron population of pod allocator jobs (in a later PR).

Reply:
Won't we be doing a lot of duplicate/extra work on the app servers? Or are we expecting these allocation jobs to happen so fast that two or more app servers picking up duplicate container tasks that need to be allocated will be rare?